### DISCUSSION TITLE
Tests fail when running multiple at the same time

### DISCUSSION THREAD

**Discussion by Mizkei1** [MikeSchulze/gdUnit4#379] [Q&A]
Hi, I have very basic setup for testing if the player can jump. The issue with this is that if I run the "PlayerCanJump" alone, it will pass just fine, but if I run the whole TestSuite tests, then it will fail. Is there some clean up that is needed to be done after the tests when using SceneRunner, or in general? Also to mention, if I comment out the "SetupWorks" test and run all the TestSuites (2 in total), the "PlayerCanJump" test will also fail in that case. Any idea what could be causing this or is this just a fault in my game's code or smth?
```
[TestSuite]
    public class PlayerJumpTests
    {
        ISceneRunner runner;
        Node2D player;

        [BeforeTest]
        public void RunBeforeTests()
        {
            runner = ISceneRunner.Load("res://TestScenes/PlayerControlTestScene.tscn");
            player = runner.FindChild("Player") as Node2D;
            runner.MaximizeView();
        }

        [TestCase]
        public void SetupWorks()
        {
            Assertions.AssertString(player.Name).IsEqual("Player");
        }

        [TestCase]
        public async Task PlayerCanJump()
        {
            var position = player.Position;
            runner.SimulateKeyPressed(Key.Space);
            await runner.AwaitIdleFrame();
            runner.SimulateKeyPressed(Key.Space);
            await runner.SimulateFrames(20);
            Assertions.AssertFloat(position.Y).IsGreater(player.Position.Y);
        }
    }
```

**Comment by MikeSchulze** [comment]
Hi, sorry for the late answer.
It looks good from test setup, the scene is loaded for every test case and should have the initial state.

To say more about this, I need an example to reproduce.

**Reply by shineforwu** [reply]
Can I define multiple private ISceneRunner objects within a [TestSuite], load them in the [Before] method, and destroy them all together in the [After] method, while testing different ISceneRunner objects in multiple [TestCase] methods without creating or destroying them? Is this approach feasible? Or is it only possible to have one ISceneRunner object within a [TestSuite]?

**Reply by MikeSchulze** [reply]
Yes, you can create many as you want, but creating them all on before and only using selective in a test i not a good pattern.

**Reply by shineforwu** [reply]
Does this mean that I can do this?
```
[BeforeTest] // or[ Before]
public void BeforeTests()
{
runner_A = ISceneRunner.Load(Path_A);
runner_B = ISceneRunner.Load(Path_B);
}
```
Can I load them simultaneously and use them separately?

**Reply by MikeSchulze** [reply]
yes
